src/test/resources/application-test.yml

```diff
@@ -16,3 +16,12 @@ spring:
         enabled: false
     compatibility-verifier:
       enabled: false # TODO: For unknown reasons there is currently a flagged incompatibility between Spring Cloud 2022.0.4 and Spring Boot 3.1.x.
+  datasource:
+    type: com.zaxxer.hikari.HikariDataSource
+    url: jdbc:h2:mem:tss_quartz;DB_CLOSE_DELAY=-1;IGNORECASE=TRUE;MODE=MySQL
+    name:
+    username: sa
+    password:
+  quartz:
+    jdbc:
+      initialize-schema: always
```

Comment:
As far as I can see, it needs 'always' so that the H2 database is set up right for the tests. Maybe it would be better not to use H2 and just stick with the testcontainer MySQL db - though that would be slower I think? I'm not sure: whatever you prefer1 😁

Reply:
We use Flyway with H2 in other places, presumably if the tests are failing without `always` then flyway isn't working with the H2 db?

Reply:
I have stripped this out and fixed the tests with some MockBean action, but I'm not sure if that's ok (I'm a little distracted with kids bouncing around for the last day of half-term)